Core/Graphics/CMakeLists.txt

```diff
@@ -11,14 +11,71 @@ set(SOURCES
     "Source/GraphicsImpl.cpp"
     "Source/GraphicsImpl.h"
     "Source/SafeTimespanGuarantor.cpp"
-    "Source/SafeTimespanGuarantor.h")
+    "Source/SafeTimespanGuarantor.h"
+    )
+
+if (WINDOWS_STORE)
+    if (BABYLON_NATIVE_WINDOWS_STORE_USE_SWAPCHAINPANEL)
+        set(SOURCES
+            ${SOURCES}
+            "Include/${BABYLON_NATIVE_PLATFORM}/SwapChainPanel/Babylon/GraphicsPlatform.h"
+            "Source/${BABYLON_NATIVE_PLATFORM}/SwapChainPanel/GraphicsPlatform.cpp")
+    else()
+        set(SOURCES
+            ${SOURCES}
+            "Include/${BABYLON_NATIVE_PLATFORM}/CoreWindow/Babylon/GraphicsPlatform.h"
+            "Source/${BABYLON_NATIVE_PLATFORM}/CoreWindow/GraphicsPlatform.cpp")
+    endif()
+elseif(APPLE)
+    if(IOS)
+        set(SOURCES
+            ${SOURCES}
+            "Include/${BABYLON_NATIVE_PLATFORM}/Babylon/GraphicsPlatform.h"
+            "Source/${BABYLON_NATIVE_PLATFORM}/iOS/GraphicsPlatform.mm")
+    else()
+        set(SOURCES
+            ${SOURCES}
+            "Include/${BABYLON_NATIVE_PLATFORM}/Babylon/GraphicsPlatform.h"
+            "Source/${BABYLON_NATIVE_PLATFORM}/macOS/GraphicsPlatform.mm")
+    endif()
+else()
+    set(SOURCES
+        ${SOURCES}
+        "Include/${BABYLON_NATIVE_PLATFORM}/Babylon/GraphicsPlatform.h"
+        "Source/${BABYLON_NATIVE_PLATFORM}/GraphicsPlatform.cpp")
+endif()
 
 add_library(Graphics ${SOURCES})
 warnings_as_errors(Graphics)
 
 target_include_directories(Graphics
-    PUBLIC "Include"
-    PRIVATE "Source/${BABYLON_NATIVE_PLATFORM}")
+            PUBLIC "Include/Shared"
+            PRIVATE "Source/${BABYLON_NATIVE_PLATFORM}")
+
+if(WINDOWS_STORE)
+    if(BABYLON_NATIVE_WINDOWS_STORE_USE_SWAPCHAINPANEL)
+        target_include_directories(Graphics
+            PUBLIC "Include/${BABYLON_NATIVE_PLATFORM}/SwapChainPanel")
```

Comment:
Im sure it would condense some of the logic here, but at the cost of readibility. I'm hesitant to make this change, as here we're pretty clear what we're doing based on build flavor and option.